### ISSUE TITLE
Error Message when syncing with ACD

### ISSUE THREAD

**Issue by c64bob** [ddurdle/PERL-CloudSync#5]
Hi,

I am syncing a folder on my GDrive with my Amazon Cloud Drive. It run perfectly for a couple of hours, then exited; the last lines of output were these:

backoff retry 1
backoff retry 2
backoff retry 3
backoff retry 4
backoff retry 5
backoff retry 6
Can't use string ("0") as a SCALAR ref while "strict refs" in use at lib/amazonapi.pm line 740, <$userInput> line 4.

**Comment by ddurdle** [comment]
When you retry, does it fail at the same folder?

**Comment by c64bob** [comment]
No, there were no more errors like this. It has been running without problems, syncing around 1TB already.

**Comment by ddurdle** [comment]
What I have noticed is that I sometimes would run into this issue and it wasn't just with the Amazon Cloud Drive.   Except for One Drive, the other services I cache folder IDs so as not to make repeated calls traversing folder paths when the folder ID is already known.  If the folder is deleted (not simply trashed) on the server and the script tries to use a cache folder ID (trying to upload to a folder that was deleted), it will run into this error.   I had predicted the possibility of this issue by wrapping around logic the situation (create a new folder and replace the cache ID with that one), but it seems there are still some issues.  What I did with Google Drive is completely disable local folder ID caching and will make the same change for Amazon Cloud Drive until it works reliably.
